pkgs/by-name/op/opencloud-desktop/package.nix

```
@@ -0,0 +1,46 @@
+{
+  lib,
+  cmake,
+  kdePackages,
+  fetchFromGitHub,
+  libre-graph-api-cpp-qt-client,
+  kdsingleapplication,
+  stdenv,
+}:
+stdenv.mkDerivation rec {
+  pname = "opencloud-desktop";
+  version = "1.0.0";
+  src = fetchFromGitHub {
+    owner = "opencloud-eu";
+    repo = "desktop";
+    tag = "v${version}";
```

finalAttrs is preferred
```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "opencloud-desktop";
  version = "1.0.0";
  src = fetchFromGitHub {
    owner = "opencloud-eu";
    repo = "desktop";
    tag = "v${finalAttrs.version}";
```

---

oh thank you! is there like a performance hit when its making the evaluations using rec vs finalAttrs?

---

there is a slight performance hit, but it is still preferred because it only introduces `finalAttrs` into the scope instead of the entire contents of the attrset and it functions better when overridden.

---

okay I found this issue: https://github.com/NixOS/nixpkgs/issues/315337
at least based on reading through it, looks like finalAttrs have a really nice advantage over rec in this case with mkDerivation, thanks for pointing that out :)

---

in the past month it's also become supported on rust, npm, go, etc.

---

oh that's awesome I did not know of it, thank you :)